### ISSUE TITLE
paginate(1) produces wrong SQL

### ISSUE THREAD

**Issue by TheZeroNine** [yajra/laravel-oci8#458]
Hey,
we have issues when using the paginate() method on a builder instance with pageSize = 1. It always returns the same row from the database.

We are querying the model like this:
```php
ResultsTable::where('identifier', $identifier)->orderBy('date_col', $orderBy)->paginate(1);
```

The generated sql looks like this:
```sql
SELECT * FROM (
    SELECT * FROM "RESULTS_V" where "identifier" = '<uuid here>' and exists (
        SELECT * FROM "RESULT_RELATIONSHIP" where "RESULT_V"."ID" = "RESULT_RELATIONSHIP"."ID_REF"
    ) order by "DATE_COL" asc
) where rownum = 1
```

When using pageSize = 2 everything works as expected and the SQL Statement is correct:
```sql
SELECT t2.* FROM (
    SELECT rownum AS "rn", t1.* FROM (
        SELECT * FROM "RESULTS_V" where "identifier" = '<uuid here>' and exists (
            SELECT * FROM "RESULT_RELATIONSHIP" where "RESULT_V"."ID" = "RESULT_RELATIONSHIP"."ID_REF"
        ) order by "DATE_COL" asc
    ) t1
) t2 where t2."rn" between 1 and 2
```

We got it working (nasty workaround) by always returning 
```php
"select t2.* from ( select rownum AS \"rn\", t1.* from ({$sql}) t1 ) t2 where t2.\"rn\" {$constraint}"
```
in `OracleGrammar.php -> compileTableExpression` and
```php
"between {$start} and {$finish}"
```
in `OracleGrammar.php -> compileRowConstraint`.

Any ideas on how to tackle this issue in a good way?

- Locally: Windows 7 (artisan serve), Remotely: Apache 2 on RedHat
- PHP Version: 7.1.15
- Laravel Version: 5.6.0
- Laravel-OCI8 Version: 5.6.0

**Comment by TheZeroNine** [comment]
if I can assist you in any way, just hit me up!

**Comment by yajra** [comment]
It would be a big help if you can submit a PR on your solution. I think I would also go on returning the `"between {$start} and {$finish}"` but my only concern for that is the `rn` field. 

Not sure, but it might break a current installation since additional field will be added on their api response.

**Comment by TheZeroNine** [comment]
our "solution" is just a quick hack to get the paginate() working, it will break a lot of other stuff as far as I can see.. we'll try to find a real solution and submit a pr then

**Comment by TheZeroNine** [comment]
we go with the extra `rn` field and reverted `OracleGrammar.php` to the version before you've switched to `rownum = 1` (7b877a1ca293ffd83b85e9b65a64de4072eda07e). I agree on the issue with the possibillity to break current installations when returning `rn`, but removing the field from the result has to be done somewhere in the application (e.g. `->except(['rn'])`). Another solution would be to make big changes to your package for a small edge case.

**Comment by yajra** [comment]
@TheZeroNine sent a PR. Can you verify if it works for you? I reverted back the `rn` for paginated results but retain the no `rn` result for queries like `first()` and `find()`.

**Comment by TheZeroNine** [comment]
@yajra works perfectly! We don't have any issues now! Thanks for your quick support!

**Comment by yajra** [comment]
Released on v5.7.1 🚀 thanks for checking.

**Comment by pedrosancao** [comment]
This issue still affecting version 5.6 and lower, I'm trying to merge for versions 5.5 and 5.6 but not sure how to proceed
